### PR TITLE
Add smart_proxy_shellhooks (DEB)

### PR DESCRIPTION
**Pull request by lzap**
For plugin updates, please indicate which repos this should be built into:

* Nightly
* 2.3

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

**Comment by lzap** [comment]
Counterpart for #6151

Do not merge yet, need to test this.

**Comment by lzap** [comment]
And it works:

```
root@debian10:~/gem2deb/ruby-smart-proxy-shellhooks-0.9.1# dpkg -L ruby-smart-proxy-shellhooks
/.
/etc
/etc/foreman-proxy
/etc/foreman-proxy/settings.d
/etc/foreman-proxy/settings.d/shellhooks.yml
/usr
/usr/lib
/usr/lib/ruby
/usr/lib/ruby/vendor_ruby
/usr/lib/ruby/vendor_ruby/smart_proxy_shellhooks
/usr/lib/ruby/vendor_ruby/smart_proxy_shellhooks/shellhooks.rb
/usr/lib/ruby/vendor_ruby/smart_proxy_shellhooks/shellhooks_api.rb
/usr/lib/ruby/vendor_ruby/smart_proxy_shellhooks/shellhooks_http_config.ru
/usr/lib/ruby/vendor_ruby/smart_proxy_shellhooks/version.rb
/usr/lib/ruby/vendor_ruby/smart_proxy_shellhooks.rb
/usr/share
/usr/share/doc
/usr/share/doc/ruby-smart-proxy-shellhooks
/usr/share/doc/ruby-smart-proxy-shellhooks/README.md
/usr/share/doc/ruby-smart-proxy-shellhooks/changelog.Debian.gz
/usr/share/doc/ruby-smart-proxy-shellhooks/copyright
/usr/share/foreman-proxy
/usr/share/foreman-proxy/bundler.d
/usr/share/foreman-proxy/bundler.d/shellhooks.rb
/usr/share/rubygems-integration
/usr/share/rubygems-integration/all
/usr/share/rubygems-integration/all/specifications
/usr/share/rubygems-integration/all/specifications/smart_proxy_shellhooks-0.9.1.gemspec
/var
/var/lib
/var/lib/foreman-proxy
/var/lib/foreman-proxy/shellhooks
/var/lib/foreman-proxy/shellhooks/README
/var/lib/foreman-proxy/shellhooks/ansible-tower-cli
/var/lib/foreman-proxy/shellhooks/print_args
/var/lib/foreman-proxy/shellhooks/print_body
root@debian10:~/gem2deb/ruby-smart-proxy-shellhooks-0.9.1# curl http://localhost:8000/shellhook/
{"valid":["print_args","print_body","ansible-tower-cli"],"invalid":["README"]}
```

File: plugins/smart_proxy_shellhooks/changelog [hunk 1]
```diff
@@ -0,0 +1,5 @@
+ruby-smart-proxy-shellhooks (0.9.1-1) UNRELEASED; urgency=medium
```
Comment (evgeni) [inline plugins/smart_proxy_shellhooks/changelog:1]:
```suggestion
ruby-smart-proxy-shellhooks (0.9.1-1) stable; urgency=medium
```

File: plugins/smart_proxy_shellhooks/changelog [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+  * Initial release (Closes: #nnnn)
```
Comment (evgeni) [inline plugins/smart_proxy_shellhooks/changelog:3]:
```suggestion
  * Initial release
```

File: plugins/smart_proxy_shellhooks/control [hunk 3]
```diff
@@ -0,0 +1,24 @@
+Source: ruby-smart-proxy-shellhooks
+Section: ruby
+Priority: optional
+Maintainer: Debian Ruby Extras Maintainers <pkg-ruby-extras-maintainers@lists.alioth.debian.org>
+Uploaders: Lukas Zapletal <lzap+pub@redhat.com>
```
Comment (evgeni) [inline plugins/smart_proxy_shellhooks/control:5]:
```suggestion
Maintainer: Lukas Zapletal <lzap+pub@redhat.com>
```

File: plugins/smart_proxy_shellhooks/control [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+Build-Depends: debhelper (>= 11~),
+               gem2deb
+Standards-Version: 4.3.0
+Vcs-Git: https://salsa.debian.org/ruby-team/ruby-smart-proxy-shellhooks.git
+Vcs-Browser: https://salsa.debian.org/ruby-team/ruby-smart-proxy-shellhooks
```
Comment (evgeni) [inline plugins/smart_proxy_shellhooks/control:10]:
```suggestion
```

**Review by evgeni — CHANGES_REQUESTED** [review]
minor nitpicks, ack as soon as adressed

**Comment by lzap** [comment]
I squashed it all.

File: plugins/smart_proxy_shellhooks/control [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+Homepage: http://github.com/theforeman/smart_proxy_shellhooks
+Testsuite: autopkgtest-pkg-ruby
+XS-Ruby-Versions: all
+Rules-Requires-Root: no
+
+Package: ruby-smart-proxy-shellhooks
+Architecture: all
+XB-Ruby-Versions: ${ruby:Versions}
+Depends: ruby | ruby-interpreter,
+         foreman-proxy (>= 2.20.0),
```
Comment (evgeni) [inline plugins/smart_proxy_shellhooks/control:17]: your tardis is broken, we don't have 2.20 yet.

**Review by evgeni — CHANGES_REQUESTED** [review]

File: plugins/smart_proxy_shellhooks/control (hunk 5 above)
Comment (lzap) [reply plugins/smart_proxy_shellhooks/control]: Hmmm, thanks. Rebased.

**Review by evgeni — APPROVED** [review]

**Pull request merged**